src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -539,22 +545,165 @@ impl HeavyHittersData {
     }
 }
 
+/// The linearized log lines for a single span. The buffer is used to accumulate partial lines until
+/// they are completed.
+#[derive(Default)]
+struct SpanLogLines {
+    /// The completed lines for this span.
+    lines: Vec<Vec<u8>>,
```

Comment:
Did you consider whether this should/could support removing lines that won't be displayed to manage peak memory usage/not buffer all the output? E.g. `VecDeque<Vec<u8>>` and old/irrelevant lines get popped off (or, with my other comment about just using `buffer`, something like `buffer: VecDeque<u8>`, although the fanciest might not be worth the performance).